### PR TITLE
TBTC documentation: deposit and redemption overview plus small reorg

### PR DESCRIPTION
**Pull request by pdyraga**
Refs #144 

[Added an overview of depositing and redeeming](https://github.com/keep-network/tbtc-v2/commit/d80d7bd55ade0abdcc4ff3067dc38c5c097bae91) so that everyone can get an overview of how the system works without a deep-dive into The Mechanism. [Moved Upgradeability and Wallet States to the end of architecture section](https://github.com/keep-network/tbtc-v2/commit/87332b35f7330134cfaad83b14abb36f6f4b9d60). It makes sense to discuss Bridge, Bank, and Vault first and have the reader understand these concepts before anything else.

### CODE REVIEW

File: docs/index.adoc [hunk 1]
```diff
@@ -30,45 +30,29 @@ anyone.
...
-=== Wallet States
+=== Deposit
 
-There are six states for a wallet:
+Stakers periodically create 51-of-100 threshold-ECDSA-backed wallets to hold
+frozen Bitcoin assets to maintain account balances. Depositors send Bitcoin to
+the most-recently-created wallet by using pay-to-script-hash (P2SH) or
+pay-to-witness-script-hash (P2WSH) which contains hashed information about the
+depositor’s Ethereum address. The depositor <<depositing,reveals>> their P2(W)SH
+deposit to Ethereum chain. The Bridge listens for these sorts of messages and
```
Comment (michalinacienciala) [inline docs/index.adoc:42]:
```suggestion
deposit to the Ethereum chain. The Bridge listens for these sorts of messages and
```

File: docs/index.adoc [hunk 2]
```diff
@@ -30,45 +30,29 @@ anyone.
...
+when it gets one, it checks the Bitcoin network to make sure the funds line up.
+If everything checks out, wallet <<sweeping,sweeps>> funds and updates
+Ethereum-side account balances.
```
Comment (michalinacienciala) [inline docs/index.adoc:45]: "the" x3?
```suggestion
If everything checks out, the wallet <<sweeping,sweeps>> the funds and updates
the Ethereum-side account balances.
```

**Review by michalinacienciala — COMMENTED** [review]
Only two grammar-related comments.

File: docs/index.adoc (hunk 1 above)
Comment (pdyraga) [reply docs/index.adoc]: 9f853d8204fbcc817b8b316446f78b1a41806977

File: docs/index.adoc (hunk 2 above)
Comment (pdyraga) [reply docs/index.adoc]: I am not sure about the second *the* (@beaushinkle ?) but added the other two: 9f853d8204fbcc817b8b316446f78b1a41806977.

Comment (beaushinkle) [reply docs/index.adoc]: "the wallet sweeps the funds and updates the balances" is better than "the wallet sweeps funds and updates the balances"

Comment (beaushinkle) [reply docs/index.adoc]: I'm not going to block on a debatable "the" though. In we go!

**Review by beaushinkle — APPROVED** [review]

**Pull request merged**